pkg/lib/localstore/ftsearch/autobatchertantivy.go

```diff
@@ -0,0 +1,131 @@
+package ftsearch
+
+import (
+	"fmt"
+
+	"github.com/anyproto/tantivy-go/go/tantivy"
+	"github.com/blevesearch/bleve/v2/search"
+	"github.com/valyala/fastjson"
+)
+
+const docLimit = 10000
+
+func (f *ftSearch2) NewAutoBatcher(maxDocs int, maxSizeBytes uint64) AutoBatcher {
+	return &ftIndexBatcher2{
+		index: f.index,
+	}
+}
+
+func (f *ftSearch2) Iterate(objectId string, fields []string, shouldContinue func(doc *SearchDoc) bool) (err error) {
+	result, err := f.index.Search(fmt.Sprintf("%s:%s", fieldId, objectId), docLimit, false, fieldId)
+	if err != nil {
+		return err
+	}
+
+	var p fastjson.Parser
```

Comment:
you also could use https://pkg.go.dev/github.com/valyala/fastjson#ParserPool

Reply:
done